lms/djangoapps/instructor/tests/test_legacy_raw_download_csv.py

```diff
@@ -104,6 +107,147 @@ def test_grade_summary_data(self):
+
+        for index, student in enumerate(expected_data['students']):
+            self.assertEqual(
+                student.username,
+                data['students'][index].username
+            )
+            self.assertListEqual(
+                expected_data['data'][index],
+                data['data'][index]
+            )
+
+    def test_grade_summary_data_no_grades(self):
+        """
+        Test grade summary data report generation with
+        get_grades set to False.
+        """
+        request = DummyRequest()
+        self.answer_question()
+        data_sets = []
+
+        # vanilla
+        data_sets.append(get_student_grade_summary_data(
+            request, self.course, get_grades=False))
+        # get_raw_scores and use_offline do nothing if
+        # get_grades is False.
+        data_sets.append(get_student_grade_summary_data(
+            request, self.course, get_grades=False, get_raw_scores=False))
+        data_sets.append(get_student_grade_summary_data(
+            request, self.course, get_grades=False, use_offline=False))
+
+        expected_data = self.get_expected_grade_data(get_grades=False)
+        self.assertFalse("assignments" in expected_data)
+
+        for data in data_sets:
+            for index, student in enumerate(expected_data['students']):
+                self.assertEqual(
+                    student.username,
+                    data['students'][index].username
+                )
+                self.assertListEqual(
+                    expected_data['data'][index],
+                    data['data'][index]
+                )
+
+    def test_grade_summary_data_use_offline(self):
+        """
+        Test grade summary data report generation with use_offline True.
+        """
+        request = DummyRequest()
+        self.answer_question()
+        data = get_student_grade_summary_data(
+            request, self.course, use_offline=True)
+        expected_data = self.get_expected_grade_data(use_offline=True)
+
+        for key in ['assignments', 'header']:
+            self.assertListEqual(expected_data[key], data[key])
+
+        for index, student in enumerate(expected_data['students']):
+            self.assertEqual(
+                student.username,
+                data['students'][index].username
+            )
+            self.assertListEqual(
+                expected_data['data'][index],
+                data['data'][index]
+            )
+
+    def test_grade_summary_data_use_offline_and_raw_scores(self):
+        """
+        Test grade summary data report generation with use_offline
+        and get_raw_scores both True.
+        """
+        request = DummyRequest()
+        self.answer_question()
+        data = get_student_grade_summary_data(
+            request, self.course, use_offline=True, get_raw_scores=True)
+        expected_data = self.get_expected_grade_data(
+            use_offline=True, get_raw_scores=True)
```

Comment:
Nit: Closing parens on following line (as in L234-7): see https://github.com/edx/edx-platform/wiki/Python-Guidelines#breaking-long-lines